### PR TITLE
share: add RowsWithNamespace tests

### LINKED ISSUE #3430: feat(share): Add unit tests for RowsWithNamespace
### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by sueun-dev**
Closes #3430.

This adds direct unit coverage for `RowsWithNamespace` in the `share` package. The tests cover rows whose namespace range includes the target namespace, an absent namespace, and the error path when a row root is too short to decode.

Tested with:

- `go test ./share`
- `go test ./share/...`
- `git diff --check`

### CODE REVIEW

File: share/root_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"bytes"
+	"testing"
+
+	libshare "github.com/celestiaorg/go-square/v4/share"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRowsWithNamespace(t *testing.T) {
```
Comment (mcrakhman) [inline share/root_test.go:11]: Tbh, not sure if in current shape and form the tests provide a lot of value, because the purpose of tests is to make sure that we cover all the possible scenarios in which this function is used.

To mark this issue complete I would like to see:
- edge cases and tests covering that
- fuzzy tests

**Review by mcrakhman — COMMENTED** [review]
Thanks for the PR! See my comments

**Comment by vgonkivs** [comment]
He @sueun-dev, do you have time to work on @mcrakhman's proposals?
